src/app/auth/resetPassword/index.tsx

```diff
@@ -21,16 +21,22 @@ function ResetPasswordScreen() {
   const [confirmPasswordTextHidden, setConfirmPasswordTextHidden] =
     useState(true);
   const [loading, setLoading] = useState(false);
+  const [passwordIsValid, setPasswordIsValid] = useState(false);
 
   const [hasUppercase, setHasUppercase] = useState(false);
   const [hasLowercase, setHasLowercase] = useState(false);
   const [hasNumber, setHasNumber] = useState(false);
   const [hasLength, setHasLength] = useState(false);
-  const [isDifferent, setIsDifferent] = useState(false);
+  const [isDifferent, setIsDifferent] = useState(true);
   const [isMatching, setIsMatching] = useState(false);
 
+  useEffect(() => {
+    if (hasUppercase && hasLowercase && hasLength && hasNumber && isDifferent) {
+      setPasswordIsValid(true);
+    }
+  });
```

Comment:
You can add a dependency array in the useEffect so this only runs when it needs to. You can see an example [here](https://react.dev/reference/react/useEffect). In this case, the dependencies for this function are each of the variables in the if statement.